media-video/obs-studio/obs-studio-28.1.1.ebuild

```diff
@@ -0,0 +1,278 @@
+		dev-qt/qtnetwork:5
+		dev-qt/qtquickcontrols:5
+		dev-qt/qtsvg:5
+		dev-qt/qtwidgets:5
+		dev-qt/qtxml:5
+	)
+	speex? ( media-libs/speexdsp )
+	ssl? ( net-libs/mbedtls:= )
+	truetype? (
+		media-libs/fontconfig
+		media-libs/freetype
+	)
+	v4l? (
+		media-libs/libv4l
+		virtual/udev
+	)
+	vlc? ( media-video/vlc:= )
+	wayland? (
+		dev-libs/wayland
+		x11-libs/libxkbcommon
+	)
+"
+RDEPEND="${DEPEND}"
+
+QA_PREBUILT="
+	usr/lib*/obs-plugins/chrome-sandbox
+	usr/lib*/obs-plugins/libcef.so
+	usr/lib*/obs-plugins/libEGL.so
+	usr/lib*/obs-plugins/libGLESv2.so
+	usr/lib*/obs-plugins/libvk_swiftshader.so
+	usr/lib*/obs-plugins/libvulkan.so.1
+	usr/lib*/obs-plugins/swiftshader/libEGL.so
+	usr/lib*/obs-plugins/swiftshader/libGLESv2.so
+"
+
+pkg_setup() {
+	use lua && lua-single_pkg_setup
+	use python && python-single-r1_pkg_setup
+}
+
+src_unpack() {
+	default
+
+	if [[ ${PV} == 9999 ]]; then
+		local EGIT_SUBMODULES=()
+		use browser && EGIT_SUBMODULES+=( plugins/obs-browser )
+		use websocket && EGIT_SUBMODULES+=(
+			plugins/obs-websocket
+			plugins/obs-websocket/deps/asio
+			plugins/obs-websocket/deps/json
+			plugins/obs-websocket/deps/qr
+			plugins/obs-websocket/deps/websocketpp
+		)
+		git-r3_src_unpack
+	else
+		if use browser; then
+			rm -d ${P}/plugins/obs-browser || die
+			mv obs-browser-${OBS_BROWSER_COMMIT} ${P}/plugins/obs-browser || die
+		fi
+
+		if use websocket; then
+			rm -d ${P}/plugins/obs-websocket || die
+			mv obs-websocket-${OBS_WEBSOCKET_COMMIT} ${P}/plugins/obs-websocket || die
+
+			rm -d ${P}/plugins/obs-websocket/deps/asio || die
+			mv asio-${ASIO_COMMIT} ${P}/plugins/obs-websocket/deps/asio || die
+	￼
+			rm -d ${P}/plugins/obs-websocket/deps/json || die
+			mv json-${JSON_COMMIT} ${P}/plugins/obs-websocket/deps/json || die
+	￼
+			rm -d ${P}/plugins/obs-websocket/deps/qr || die
+			mv QR-Code-generator-${QR_COMMIT} ${P}/plugins/obs-websocket/deps/qr || die
+	￼
+			rm -d ${P}/plugins/obs-websocket/deps/websocketpp || die
+			mv websocketpp-${WEBSOCKETPP_COMMIT} ${P}/plugins/obs-websocket/deps/websocketpp || die
+		fi
+	fi
+}
+
+src_prepare() {
+	# Avoid errors if we don't have 'obs-browser' or 'obs-websocket'.
+	if ! use browser; then
+		sed -i 's/FATAL_ERROR "obs-browser/STATUS "obs-browser/' plugins/CMakeLists.txt || die
+	fi
+
+	if ! use websocket; then
+		sed -i 's/FATAL_ERROR "obs-websocket/STATUS "obs-websocket/' plugins/CMakeLists.txt || die
+	fi
+
+	cmake_src_prepare
+}
+
+src_configure() {
+	local libdir=$(get_libdir)
+	local mycmakeargs=(
+		$(usev browser -DCEF_ROOT_DIR=../${CEF_DIR})
+		-DENABLE_ALSA=$(usex alsa)
+		-DENABLE_AJA=OFF
+		-DENABLE_BROWSER=$(usex browser)
+		-DENABLE_DECKLINK=$(usex decklink)
+		-DENABLE_FREETYPE=$(usex truetype)
+		-DENABLE_JACK=$(usex jack)
+		-DENABLE_LIBFDK=$(usex fdk)
+		-DENABLE_NEW_MPEGTS_OUTPUT=OFF # Requires librist and libsrt.
+		-DENABLE_PIPEWIRE=$(usex pipewire)
+		-DENABLE_PULSEAUDIO=$(usex pulseaudio)
+		-DENABLE_RTMPS=$(usex ssl ON OFF)
```

Comment:
Oh right, fwiw metadata.xml needs an update if add this (currently says USE=ssl is for RTMPS only).

Reply:
Right, I forgot there were some changes regarding this one.

Reply:
The "what's new" requires browser as well.

Reply:
Unsure how to best handle "what's new" in terms of metadata, since it requires both, 'browser' and 'ssl' (right now the feature is enabled if the aforementioned flags are enabled).

Reply:
I see wrt browser. Realistically disabling USE=ssl is hardly supported in Gentoo in the first place, so I wouldn't worry too much about this as long as it builds.

Reply:
And on that note.. I don't think it's worth having:
```
 * For the "What's New" feature, the 'browser' and 'ssl'
 * USE-flags need to be enabled.
```
For everyone that happened to have USE=-browser shown every time they update for just missing a what's new page.

I'll go ahead and yank it out on merge, hope you don't mind.

Reply:
Yeah I guess it's fine to be yanked, and yeah, there were some issues and I didn't get to getting rid of the draft yet oops.